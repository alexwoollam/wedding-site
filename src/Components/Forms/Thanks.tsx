import React from 'react';
import Heading from '../Text/Heading';

function Thanks( props:any ) {
    
    const text = ({
        thanks: "Thanks, " + props.name + "!!",
    });
    
    return (
        <div>
            <Heading>{text.thanks}</Heading>
        </div>
    );
    
}

export default Thanks;