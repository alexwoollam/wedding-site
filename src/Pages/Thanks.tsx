import React from 'react';
import {connect} from "react-redux";
import {Col, Container, Row} from "reactstrap";
import Page from "../Components/Layout/Page";
import Heading from "../Components/Text/Heading";
import Content from "../Content/Pages/RSVP.json";
import OtherContent from "../Content/Forms/Rsvp.json";
import Bugs from "../Components/Bugs";
import TextBlock from "../Components/Text/TextBlock";

const Thanks = () => {
    return (
        <Page>
            <Row>
                <Col>
                    <Bugs/>
                    <Heading main subtitle={Content.subtitle}>{Content.title}</Heading>
                </Col>
            </Row>
            <Row>
                <Container>
                    <Heading main>{OtherContent.step_3.title}</Heading>
                    <Container>
                        <TextBlock>{OtherContent.step_3.accommodation_bumph_one}</TextBlock>
                        <TextBlock>{OtherContent.step_3.accommodation_bumph_two}</TextBlock>
                        <TextBlock>{OtherContent.step_3.accommodation_bumph_three}</TextBlock>
                        <a href={OtherContent.step_3.accommodation_bumph_link}>{OtherContent.step_3.accommodation_bumph_link_text}</a>
                    </Container>
                </Container>
            </Row>
        </Page>
    );
}

const connected = connect(
    null,
    null,
)(Thanks);
export { connected as Thanks };