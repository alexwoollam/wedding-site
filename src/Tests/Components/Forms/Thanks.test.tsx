import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Thanks from '../../../Components/Forms/Thanks';

describe('<Thanks name="name" /> component', () => {
    test('Thanks block name should return correct props.', () => {
    render(<Thanks name='foo bar-bundy' />);
    const ThanksBlockContent = screen.getByText('Thanks, foo bar-bundy!!');
    expect(ThanksBlockContent).toBeInTheDocument();
  });
});
