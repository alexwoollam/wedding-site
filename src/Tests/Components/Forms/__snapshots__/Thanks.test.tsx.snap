// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thanks name="name" /> component renders correctly 1`] = `
<div>
  <div
    className="heading"
  >
    <h1
      className="text-grey-700 font-serif text-3xl text-center heading-text"
      data-testid="text-block"
    >
      Thanks, undefined!!
    </h1>
    <span
      className="heading-subtext"
    />
  </div>
</div>
`;
